src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests_DefaultsSource.vb

```diff
@@ -0,0 +1,128 @@
+                Await state.AssertSelectedCompletionItem("MyAB", isHardSelected:=True)
+            End Using
+        End Function
+
+        <WpfTheory, CombinatorialData>
+        <Trait(Traits.Feature, Traits.Features.Completion)>
+        Public Async Function Test3(isAggressive As Boolean) As Task
+            Using state = CreateTestState(
+                              <Document>
+using NS1;
+class C
+{
+    void Method()
+    {
+        MyA$$
+    }
+}
+                              </Document>)
+                If isAggressive Then
+                    state.TextView.Options.SetOptionValue(ItemManager.AggressiveDefaultsMatchingOptionName, True)
+                End If
+
+                state.SendInvokeCompletionList()
+
+                If isAggressive Then
+                    Await state.AssertCompletionItemsContain("MyA", displayTextSuffix:="")
+                    Await state.AssertSelectedCompletionItem("MyAB", isHardSelected:=True)
+                Else
+                    Await state.AssertCompletionItemsContain("MyAB", displayTextSuffix:="")
+                    Await state.AssertSelectedCompletionItem("MyA", isHardSelected:=True)
+                End If
+            End Using
+        End Function
+
+        Private Shared Function CreateTestState(documentElement As XElement) As TestState
+            Return TestStateFactory.CreateTestStateFromWorkspace(
+                <Workspace>
+                    <Project Language="C#" LanguageVersion="Preview" CommonReferences="true">
+                        <ProjectReference>RefProj</ProjectReference>
+                        <Document FilePath="C.cs">
+                            <%= documentElement.Value %>
+                        </Document>
+                    </Project>
+                    <Project Language="C#" AssemblyName="RefProj" CommonReferences="true">
+                        <Document><![CDATA[
+namespace NS1
+{
+    public class MyA
+    {
+    }
+    public class MyAB
+    {
+    }
+}
+                        ]]></Document>
+                    </Project>
+                </Workspace>,
+                extraExportedTypes:={GetType(MockDefaultSource)}.ToList())
+        End Function
+
+        <ComponentModel.Composition.Export(GetType(IAsyncCompletionDefaultsSource))>
+        <VisualStudio.Utilities.Name(NameOf(MockDefaultSource))>
+        <VisualStudio.Utilities.ContentType(ContentTypeNames.RoslynContentType)>
+        <TextViewRole(PredefinedTextViewRoles.Document)>
+        Private Class MockDefaultSource
+            Implements IAsyncCompletionDefaultsSource
+
+            <ComponentModel.Composition.ImportingConstructor>
+            <Obsolete(MefConstruction.ImportingConstructorMessage, True)>
+            Public Sub New()
+            End Sub
+
+            Public Function GetSessionDefaultsAsync(session As IAsyncCompletionSession) As Task(Of ImmutableArray(Of String)) Implements IAsyncCompletionDefaultsSource.GetSessionDefaultsAsync
```

Comment:
i don't really understand the term 'session defaults'.

Reply:
Those are editor APIs, and we were not consulted when they were created. We should be able to change it since it's internal though
http://index/?query=IAsyncCompletionDefaultsSource&rightProject=Microsoft.VisualStudio.Language&file=AsyncCompletion%5CIAsyncCompletionDefaultsSource.cs&line=39